autometa/common/kmers.py

```diff
@@ -319,13 +430,13 @@ def embed(kmers=None, embedded=None, n_components=2, do_pca=True, pca_dimensions
         n_samples, n_dims = X.shape
 
     logger.debug(f'{method}: {n_samples} data points and {n_dims} dimensions')
-    # Adjust perplexity according to the number of data points
-    n_rows = n_samples-1
-    scaler = 3.0
-    if n_rows < (scaler*perplexity):
-        perplexity = (n_rows/scaler) - 1
 
     def do_TSNE(perplexity=perplexity, n_components=n_components):
+        # Adjust perplexity according to the number of data points
+        n_rows = n_samples-1
+        scaler = 3.0
+        if n_rows < (scaler*perplexity):
+            perplexity = (n_rows/scaler) - 1
```

Comment:
There needs to be some documentation somewhere that this gives different results to the Python2.7 version. Perhaps in the docs or in comments somewhere.

Reply:
I'll place a note, and we can place a link to a discussion on the differences in the readthedocs documentation.